File: 005/solve.py
#!usr/bin/python

from math import sqrt

def isprime(n):
    for x in xrange(2, int(sqrt(n))+1):
        if n%x == 0:
            return False
    return True

def primes():
    y = 2
    while True:
        pr = y
        yield pr
        y += 1
        while isprime(y) is False:
            y += 1

max = 20
result = 1
for x in primes():
    K = 1
    if x > max:
        break
    while K*x < max+1:
        K *= x
    result *= K

print result

File: 005/solve.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	for i := 2; i < int(math.Sqrt(float64(n))+1); i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	primes := make(chan int)

	go func() {
		for i := 2; ; i++ {
			if isPrime(i) {
				primes <- i
			}
		}
	}()

	max := 20
	result := 1

	for prime := range primes {
		k := 1
		if prime > max {
			break
		}
		for k*prime < max+1 {
			k *= prime
		}
		result *= k
	}

	fmt.Println(result)
}
